### PR TITLE
Fixes missing objectives and antag status.

### LINKED ISSUE #29264: Traitor objectives don't always show up in the greentext at end of round
https://file.house/GYUp.png

https://file.house/P8Jr.png

This happened to two other people around 3:30 pm EST 7/14/17. 

The following pull requests are currently test merged:
#29241: 'ClickCross now uses CanPass instead of density directly.' by AnturK at commit a90688

Based off origin/master commit: 2b053f

### LINKED ISSUE #29619: Antag status removed after having player's mind/body destroyed
**Cross-posting from Citadel:**

- What happened: Player antag unintentionally removed from antag status
- How it happened: Player's body and mind were destroyed
- How to recreate it: Assign player to antag status and destroy their body via gibbing or possibly simply deleting

Antag status seems to be removed after a player (ckey "wilchen" in this case) has their body/mind destroyed. Noticed this after someone BoH bombed the super matter crystal and was gibbed. I questioned them about it as they did not appear on the Traitor Panel, but they claimed that they were indeed an antag and had completed all their objectives except the "die a glorious death" objective. I was only able to find out that they were indeed a traitor by checking the game log (see pic 1). At the end of the round they still did not appear in the end of round summary (see pic 2). Unfortunately I don't have an image of the Traitor Panel as the round ended while I was still looking into this.

**Pic 1**

**Pic 2**

Can provide further logs/info if requested.

### PR DESCRIPTION
**Pull request by AnturK**
This whole thing is still horrible.

Fixes #29264 
Fixes #29619

### CODE REVIEW

File: code/datums/antagonists/datum_traitor.dm [hunk 1]
```diff
@@ -31,16 +32,16 @@
...
 	should_give_codewords = FALSE
 	give_objectives = FALSE
 
 
 /datum/antagonist/traitor/on_body_transfer(mob/living/old_body, mob/living/new_body)
-	if(issilicon(new_body) && issilicon(old_body))
-		..()
-	else
+	// human <-> silicon only
+	if(old_body && issilicon(new_body) ^ issilicon(old_body))
```
Comment (lzimann) [inline code/datums/antagonists/datum_traitor.dm]: `^` ?

Comment (AnturK) [inline code/datums/antagonists/datum_traitor.dm]: xor, silicon to silicon or human to human don't need special cases.

Comment (lzimann) [inline code/datums/antagonists/datum_traitor.dm]: Oh right, I forgot xor is a thing.

**Review by Cyberboss — APPROVED** [review]

File: code/datums/antagonists/datum_traitor.dm [hunk 2]
```diff
@@ -83,7 +85,8 @@
 	if(owner.assigned_role == "Clown")
 		var/mob/living/carbon/human/traitor_mob = owner.current
 		if(traitor_mob&&istype(traitor_mob))
-			if(!silent) to_chat(traitor_mob, "Your training has allowed you to overcome your clownish nature, allowing you to wield weapons without harming yourself.")
+			if(!silent) 
+				to_chat(traitor_mob, "Your training has allowed you to overcome your clownish nature, allowing you to wield weapons without harming yourself.")
 			traitor_mob.dna.remove_mutation(CLOWNMUT)
```
Comment (ChangelingRain) [inline code/datums/antagonists/datum_traitor.dm]: this should probably check for the clown mutation rather than assigned role.

File: code/datums/antagonists/datum_traitor.dm (hunk 1 above)
Comment (lordpidey) [inline code/datums/antagonists/datum_traitor.dm]: Huh, this looked like it would need another set of parenthesis,

But I looked it up, and ^ has priority over &&.

Wierd, I thought they had same priority.

File: code/datums/antagonists/datum_traitor.dm (hunk 2 above)
Comment (Cyberboss) [inline code/datums/antagonists/datum_traitor.dm]: Eh, can do that in a different PR
